File: builder/java/LessonDirector.java
package Builder;

// Builder arayüzünü kullanarak nesne örneklemesi yapar.
// UML diyagramındaki Director'a denk gelmektedir.
public class LessonDirector {
    private LessonBuilder lessonBuilder;

    public LessonDirector(LessonBuilder lessonBuilder) {
        this.lessonBuilder = lessonBuilder;
    }

    public void make() {
        lessonBuilder.getLesson();
        lessonBuilder.applyDiscount();
        lessonBuilder.addLessonNote();
    }
}

File: builder/typescript/lesson-director.ts
import {LessonBuilder} from "./lesson-builder";

// Builder arayüzünü kullanarak nesne örneklemesi yapar.
// UML diyagramındaki Director'a denk gelmektedir.
export class LessonDirector {
  constructor(private readonly lessonBuilder: LessonBuilder) {}

  public make(): void {
    this.lessonBuilder.getLesson();
    this.lessonBuilder.applyDiscount();
    this.lessonBuilder.addLessonNote();
  }
}
